[3006.x] auto set gitbase defaults

cmcmarrow opened the pull request:
### What does this PR do?
remove static args for gitfs

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/65096

### Previous Behavior
Need it to provide static args for child classes

### New Behavior
No static args and defaults dont lead to crash

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

cmcmarrow committed:
auto set gitbaase defaults

cmcmarrow commented:
TODO remove pre-commit upgrade, clean up init remote, fix pylint master

twangboy reviewed (commented):
It's probably gonna take some work to get this rebased. @dmurphy18 has done a lot of work in gitfs recently.

dwoz committed:
Add changelog and fix tests for auto gitbase defaults

Add changelog/65310.changed.md documenting the deprecation of
per_remote_overrides/per_remote_only/global_only arguments.

Update test opts dicts in test_gitfs.py to include the gitfs keys
now required by GITFS_PER_REMOTE_OVERRIDES validation (mountpoint,
saltenv_whitelist, saltenv_blacklist, disable_saltenv_mapping,
ref_types, update_interval).
